editor/animation_track_editor.cpp

```diff
@@ -7314,19 +7333,30 @@ void AnimationTrackEditor::_selection_changed() {
 }
 
 void AnimationTrackEditor::_update_snap_unit() {
+	nearest_fps = 0;
+
 	if (step->get_value() <= 0) {
 		snap_unit = 0;
+		_update_nearest_fps_label();
 		return; // Avoid zero div.
 	}
 
 	if (timeline->is_using_fps()) {
+		_clear_selection(true); // Needs to recreate a spinbox of the KeyEdit.
```

Comment:
Hello, can you explain/document this line better? It's causing the animation player to unselect keyframes when I click them in my fork. What is this line trying to solve? Thanks!

Reply:
The FPS in the key edit inspector is displayed dependent on the FPS of the snap value. For example, a key placed at 1s will appear as follows in FPS mode:

![image](https://github.com/user-attachments/assets/01420a4b-590c-4e56-9f8f-9d5ad8aa2afa) ![image](https://github.com/user-attachments/assets/9c01a086-2a40-4a20-9399-a4fdf621befe)

I thought that changing the FPS with the keyedit inspector displayed would implicitly change the FPS in the inspector and have some negative effect, but sorry I did not realize that _update_snap_unit() is fired each key time is changed in the inspector (although I think it's a redundant).

There are some instability issues, such as undo after switching from second mode and keys not working due to spin box arrows, but this seems to be a problem that has been occurring for some time and probably needs to be revert in this area and stabilized from other parts.